### DISCUSSION TITLE
How to write relationship in models where belongs to have condition of parent model?

### DISCUSSION THREAD

**Discussion by sanmayad** [phalcon/cphalcon#15371] [Q&A]
So I have a DB doc_submitted which has two columns doc_type and doc_id where doc_type can refer to any table viz: assignment or task or book and doc_id will have respective id.

Now I want to write relationship in the DocSubmitted Model so I can refer them in view as doc_submitted.Assignment.topic or doc_submitted.Task.date.

If I'm putting the condition it's always searching in reference model instead of parent model.

**Comment by quasipickle** [comment]
I've done something similar, but not exactly the same.  In my setup, the name of the related record doesn't change, so `DocSubmitted.Type` could reference an `Assignment` record, or a `Task` record, or whatever you want.  However, `Type` is not actually a relationship. -it's a variable that is magically set via `__get()` when `DocSubmitted.Type` is referenced.

It works kind of like this (caching & error checking removed for simplicity):

```php
class DocSubmitted{
    public function __get($name){
        return ($name == 'Type') ? $this->retrieveTypeRecord() : return parent:__get($name);    
    }

    private function retrieveTypeRecord(){
        switch($this->doc_type){
            case 'Assignment':
                return Assignment::findFirst($this->doc_id);
            case 'Task':
                return Task::findFirst($this->doc_id);
        }
    }
}
```
    

This works fine for the view, where you're just reading the information.  It gets trickier when you're saving or setting `Type`.
